features/dd-sdk-android-trace/src/main/java/com/datadog/opentracing/PendingTrace.java

```diff
@@ -178,25 +180,27 @@ public void cancelContinuation(final ContinuableScope.Continuation continuation)
   }
 
   private void expireReference() {
-    final int count = pendingReferenceCount.decrementAndGet();
-    if (count == 0) {
-      write();
-    } else {
-      if (tracer.getPartialFlushMinSpans() > 0 && size() > tracer.getPartialFlushMinSpans()) {
-        synchronized (this) {
-          if (size() > tracer.getPartialFlushMinSpans()) {
-            final DDSpan rootSpan = getRootSpan();
-            final List<DDSpan> partialTrace = new ArrayList(size());
-            final Iterator<DDSpan> it = iterator();
-            while (it.hasNext()) {
-              final DDSpan span = it.next();
-              if (span != rootSpan) {
-                partialTrace.add(span);
-                completedSpanCount.decrementAndGet();
-                it.remove();
+    synchronized(this) {
+      final int count = pendingReferenceCount.decrementAndGet();
+      if (count == 0) {
+        write();
+      } else {
+        if (tracer.getPartialFlushMinSpans() > 0 && size() > tracer.getPartialFlushMinSpans()) {
+          synchronized (this) {
```

Comment:
we probably don't need this one, because we are already in `synchronized` block. But we may want to keep it although to simplify viewing the diff with `dd-trace-java`.